src/engine/FileDropHandler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   Handles signal from Godot when files are dragged and dropped onto the game window
+/// </summary>
+public class FileDropHandler : Node
+{
+    public override void _Ready()
+    {
+        GetTree().Connect("files_dropped", this, "OnFilesDropped");
+    }
+
+    private void OnFilesDropped(string[] files, int screen)
+    {
+        foreach (var file in files)
+        {
+            GD.Print("Detected file drop \"", file, "\" on screen ", screen);
+            HandleFileDrop(file, screen);
+        }
+    }
+
+    private void HandleFileDrop(string file, int screen)
+    {
+        // Currently nothing depends on this (this might be the game window index, at least on Linux this doesn't seem
+        // to change when the monitor is changed the drag happens on...)
+        _ = screen;
+
+        // For now just the load save functionality is done, but in the future this might be extended to allow
+        // other code to dynamically register listeners here
+
+        if (file.EndsWith(Constants.SAVE_EXTENSION, StringComparison.InvariantCulture))
+        {
+            if (!file.StartsWith(Constants.EXPLICIT_PATH_PREFIX, StringComparison.InvariantCulture))
+                file = Constants.EXPLICIT_PATH_PREFIX + file;
+
+            // TODO: could add a dialog to ask if the save should be loaded now or copied to the saves folder
+            HandleLoadSaveFromDrop(file);
+        }
+        else
+        {
+            GD.Print("Unknown file type to handle on drop");
+        }
+    }
+
+    private void HandleLoadSaveFromDrop(string file)
+    {
+        GD.Print("Trying to load dropped save file...");
+
+        // TODO: would be nice to have some kind of popup box showing the errors from here to the user
+        try
+        {
+            var info = Save.LoadJustInfoFromSave(file);
+
+            if (info.Type == SaveInformation.SaveType.Invalid)
```

Comment:
I attempted to drop corrupted save file but seems like this doesn't get triggered, debu log only shows exception from `LoadJustInfoFromSave`. Not very critical issue though, just thought worth mentioning.

Reply:
In my testing I found that at one point I got an invalid save type. I think LoadJustInfoFromSave is made to not throw an exception, and instead just returns an invalid save info object, or I may have used a different save load method before that does return invalid save info...

Reply:
Just checked again and yeah I must have mistaken `PrintErr` in `LoadJustInfoFromSave`'s catch block for an exception. Actually now after testing this again this seems to work correctly, I may have confused the debug log messages at some point...